src/general/OptionsMenu.cs

```diff
@@ -952,6 +994,11 @@ private void OnTutorialsEnabledToggled(bool pressed)
         UpdateResetSaveButtonState();
     }
 
+    private void InitInputGroupList()
+    {
+        InputGroupList.Instance.InitFromData(Settings.Instance.CurrentControls);
```

Comment:
but I cant make methods like ShowInputConflictDialog static. How should I call this method from InputEventItem?

Reply:
I hope my commit is what you mean

Reply:
Sorry for not being able to reply earlier, I had some other stuff to do.
You didn't do what I had in mind, it's a bit better to have static methods than a global object, but I feel like you just basically converted the global object to a static class. So it isn't much better.

Even if there is only ever one of the input group things at once, I think it's good to ask "should the input list access the global group object *or* should it access the group object that owns it?" To me it is obvious that the second one is right.

So this is the kind of class structure I recommend:

```
class InputGroupThing{
public void BuildGUI(){
var child = blah.Instance();
child.GroupOwner = this;
child.BuildGUI();
}
}

class FirstLevelChild{
public InputGroupThing {get; set;}

public void BuildGUI(){
var anotherChild = bleh.Instance();
anotherChild.GroupOwner = InputGroupThing;
anotherChild.BuildGUI();
}
}

```
That's what I meant when I said to flow / pass the input group reference through the object tree.

Reply:
Ok, I think I get what you mean.
Please check my commit and check if I did finally understand what you meant :D

Reply:
You created the GUI Control tree like I suggested, but I don't see you removing the pseudo-global usage?

Reply:
I have to do some pseudo global access, but I reduced it drasticly. Is this ok now?

Reply:
It's better now.

Related to the new code, I'm not sure I agree with how much null checks there are and ? usage. Shouldn't the GUI for key changing be always in a valid state? Having those ignore if null, could make further development much harder, if they mask null problems after refactoring or something.

Reply:
There shouldn't be any nulls, but if the weak reference was disposed, there would be null pointers everywhere otherwise. I think its safer to check if the weak reference is disposed (and my methods therefore returns null) instead of just blindly relying on it not being disposed.

Reply:
I think it should *never* be the case that the children of an input group are destroyed *after* the group. So the problem condition should never happen. For that reason I kinda would like to have the code just go ahead and flat out crash with null pointer or disposed exception, as when refactoring or doing other changes to the input binding, it will be faster to get exceptions when you do something wrong.
@Kasterisk might be able to weigh in on this.

Reply:
I'm afraid I wouldn't be much help with the workings of the code, but I'll just go with hhyyrylainen on this.